### PR TITLE
:running: Verification of leak goroutine on controller and manager

### PR DESCRIPTION
**Pull request by knabben**
This PR adds tests both for manager and controller checking the leak of goroutines on start and stop, I put the threshold of 5 in comparison because it has 3 leaks after stop:

```
goroutine 29 [chan receive]:
k8s.io/apimachinery/pkg/watch.(*Broadcaster).loop(0xc00021fa80)

goroutine 166 [chan receive]:
k8s.io/client-go/tools/record.(*eventBroadcasterImpl).StartEventWatcher.func1(0x291e580, 0xc000212ab0, 0xc000212a80)

goroutine 167 [chan receive]:
k8s.io/client-go/tools/record.(*eventBroadcasterImpl).StartEventWatcher.func1(0x291e580, 0xc000212ae0, 0xc0003fb350)
```

Related with #724

### CODE REVIEW

**Comment by alenkacz** [comment]
Hmm are the leaked goroutines expected? Or should the test rather not have a treshold of 5 but rather 0 and we should fix those leaks before landing this?

**Comment by DirectXMan12** [comment]
Those are semi-expected.  We don't have a good way to work around them yet.  I *think* we should finally have a `shutdown` method, but that doesn't alleviate the problem entirely.  Ideas welcome.

For now, the threshold should be whatever exists currently (so the three), with a comment to ratchet down as we fix the issues.

File: pkg/controller/controller_test.go [hunk 1]
```diff
@@ -90,6 +92,26 @@ var _ = Describe("controller.Controller", func() {
 
 			close(done)
 		})
+
+		It("should not leak goroutines when stop", func(done Done) {
+			// TODO(community) - After closing the proper leaks on watch this must be reduced to 0
```
Comment (alenkacz) [inline pkg/controller/controller_test.go]: should we maybe also document why are those leaks expected and what goroutines are leaking for future reference?

Comment (DirectXMan12) [reply pkg/controller/controller_test.go]: yeah.  We can at least link to the relevant issue somewhere around here.

**Comment by DirectXMan12** [comment]
gonna push the note, otherwise we should be good

File: pkg/controller/controller_test.go (hunk 1 above)
Comment (DirectXMan12) [reply pkg/controller/controller_test.go]: (also, for future reference, `community` is not actually a valid TODO target -- this isn't supposed to be who does the TODO, it's supposed to be who has context)

**Pull request merged**